server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java

```diff
@@ -88,16 +91,16 @@ public void testHighWatermarkNotExceeded() throws Exception {
                 .put(INDEX_STORE_STATS_REFRESH_INTERVAL_SETTING.getKey(), "0ms")
                 .build()
         );
-        final long minShardSize = createReasonableSizedShards(indexName);
+        var smallestShard = createReasonableSizedShards(indexName);
 
         // reduce disk size of node 0 so that no shards fit below the high watermark, forcing all shards onto the other data node
         // (subtract the translog size since the disk threshold decider ignores this and may therefore move the shard back again)
-        getTestFileStore(dataNodeName).setTotalSpace(minShardSize + WATERMARK_BYTES - 1L);
+        getTestFileStore(dataNodeName).setTotalSpace(smallestShard.getValue() + WATERMARK_BYTES - 1L);
         assertBusyWithDiskUsageRefresh(dataNode0Id, indexName, empty());
 
         // increase disk size of node 0 to allow just enough room for one shard, and check that it's rebalanced back
-        getTestFileStore(dataNodeName).setTotalSpace(minShardSize + WATERMARK_BYTES + 1L);
-        assertBusyWithDiskUsageRefresh(dataNode0Id, indexName, hasSize(1));
+        getTestFileStore(dataNodeName).setTotalSpace(smallestShard.getValue() + WATERMARK_BYTES + 1L);
+        assertBusyWithDiskUsageRefresh(dataNode0Id, indexName, contains(smallestShard.getKey()));
```

Comment:
I think this is too strong - we might have two shards of the same (smallest) size and in that case we might choose to assign either.

Reply:
Good point. Updated.